### DISCUSSION TITLE
Yarn contraints alternative for pnpm-based monorepo?

### DISCUSSION THREAD

**Discussion by guillaumewuip** [pnpm/pnpm#4016] [Q&A] [answered]
Hello pnpm world 👋

I'm looking for suggestions of tools to implement some rules over a pnpm-based monorepo. I know yarn has [contraints](https://yarnpkg.com/features/constraints) and I think I'm looking for something similar. I may not need something that generic and powerful though.

Some use cases :
- enforce we have the same version of react in every package
- enforce all packages have some obligatory scripts (eg. lint, test, build)
- enforce all packages are `private: true`
- etc.

Is this ringing bells? I'm really struggling find tools that could help. Thanks in advance 🙏

**Comment by zkochan** [comment, marked as answer]
See related issue https://github.com/pnpm/pnpm/issues/2713

I currently recommend using [syncpack](https://www.npmjs.com/package/syncpack) and [@pnpm/meta-updater](https://github.com/pnpm/meta-updater)

**Reply by guillaumewuip** [reply]
Thanks @zkochan! With those two packages I think I can write the rules I need 🙏

**Comment by m1heng** [comment]
To expand this topic, is it suitable to make this functionality of meta-updater delivered along with pnpm ?

**Reply by zkochan** [reply]
Maybe at some point yes. I think it is not useful enough yet.

**Comment by guillaumewuip** [comment]
For people coming here, I've also found https://github.com/tclindner/npm-package-json-lint that could help.
